services/provider/server/server.go

```diff
@@ -2076,3 +2081,16 @@ func getKubeResourcesForClass[T CommonClassSpecAccessors](
 	}
 	return kubeResources
 }
+
+func checkConsumerPreConditions(consumer *ocsv1alpha1.StorageConsumer, serverVersion string) error {
```

Comment:
You are checking the client's preconditions not the consumers
```suggestion
func checkClientPreConditions(consumer *ocsv1alpha1.StorageConsumer, serverVersion string) error {
```

Reply:
done.